tests/azure_iot_provisioning_client_ut.c

```diff
@@ -0,0 +1,704 @@
+                                                       &ucIdScope[ 0 ], sizeof( ucIdScope ),
+                                                       &ucRegistrationId[ 0 ], sizeof( ucRegistrationId ),
+                                                       NULL, ucBuffer, sizeof( ucBuffer ),
+                                                       prvGetUnixTime,
+                                                       &xTransportInterface ),
+                      eAzureIoTProvisioningSuccess );
+
+    AzureIoTProvisioningClient_Deinit( &xTestProvisioningClient );
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTProvisioningClient_SymmetricKeySet_Failure( void ** ppvState )
+{
+    AzureIoTProvisioningClient_t xTestProvisioningClient;
+
+    ( void ) ppvState;
+
+    /* Fail AzureIoTProvisioningClient_SetSymmetricKey when null symmetric key is passed */
+    assert_int_not_equal( AzureIoTProvisioningClient_SetSymmetricKey( &xTestProvisioningClient,
+                                                                      NULL, 0,
+                                                                      prvHmacFunction ),
+                          eAzureIoTProvisioningSuccess );
+
+    /* Fail AzureIoTProvisioningClient_SetSymmetricKey when null hashing function is passed */
+    assert_int_not_equal( AzureIoTProvisioningClient_SetSymmetricKey( &xTestProvisioningClient,
+                                                                      ucSymmetricKey, sizeof( ucSymmetricKey ) - 1,
+                                                                      NULL ),
+                          eAzureIoTProvisioningSuccess );
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTProvisioningClient_SymmetricKeySet_Success( void ** ppvState )
+{
+    AzureIoTProvisioningClient_t xTestProvisioningClient;
+
+    ( void ) ppvState;
+
+    assert_int_equal( AzureIoTProvisioningClient_SetSymmetricKey( &xTestProvisioningClient,
+                                                                  ucSymmetricKey, sizeof( ucSymmetricKey ) - 1,
+                                                                  prvHmacFunction ),
+                      eAzureIoTProvisioningSuccess );
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTProvisioningClient_Register_ConnectFailure( void ** ppvState )
+{
+    AzureIoTProvisioningClient_t xTestProvisioningClient;
+    AzureIoTProvisioningClientResult_t xResult;
+
+    ( void ) ppvState;
+
+    prvSetupTestProvisioningClient( &xTestProvisioningClient );
+
+    /* Connect */
+    will_return( prvHmacFunction, 0 );
+    will_return( AzureIoTMQTT_Connect, eAzureIoTMQTTServerRefused );
+    will_return( AzureIoTMQTT_ProcessLoop, eAzureIoTMQTTSuccess );
+    xResult = AzureIoTProvisioningClient_Register( &xTestProvisioningClient, 0 );
+    assert_int_not_equal( xResult, eAzureIoTProvisioningPending );
```

Comment:
Could we just check against the expected error code directly?  I get there's some flexibility here, but it's a bit cleaner since it's one line and not 2 for check, and we tend to pin the behavior a bit better (so if we start returning eOutOfMemory where we'd always been eFailure before from product, that'd be good to know).

Reply:
sure

---
In reply to: [607966442](https://github.com/Azure/azure-iot-middleware-freertos/pull/62#discussion_r607966442) [](ancestors = 607966442)